price-feeder/oracle/client/chain_height.go

```diff
@@ -102,5 +103,10 @@ func (chainHeight *ChainHeight) GetChainHeight() (int64, error) {
 	chainHeight.mtx.RLock()
 	defer chainHeight.mtx.RUnlock()
 
+	if chainHeight.lastChainHeight == 0 &&
+		chainHeight.errGetChainHeight == nil {
+		chainHeight.errGetChainHeight = errNoBlockHeightEvents
+	}
```

Comment:
I'm in between doing this (seems a bit lazy, but at least more clear), or adding an rpc query here to get the height (would be more effective, but make the code a bit messier) 

wdyt @RafilxTenfen ?

Reply:
I think we could start the ChainHeight with this error like 
```go

	chainHeight := &ChainHeight{
		Logger:            logger.With().Str("oracle_client", "chain_height").Logger(),
		errGetChainHeight: errNoBlockHeightEvents,
		lastChainHeight:   0,
	}

```
This way when a new block appears, it will throw the event, and from now on the `chainHeight` structure would be updated with the current block height and nil on the errGetChainHeight

Reply:
Hmm, I'm kind of leaning towards just querying for the starting height at the object initialization... that seems more clean than initializing with an error imo, and would not impact performance